backend/src/main/java/com/github/boltzmann/biokiste/backend/controller/ItemController.java

```diff
@@ -20,13 +20,22 @@ public ItemController(ItemDetailsService itemDetailsService) {
         this.itemDetailsService = itemDetailsService;
     }
 
-    @GetMapping(path="/all")
-    public List<Item> getAllItems(){
+    @GetMapping(path = "/all")
+    public List<Item> getAllItems() {
         return itemDetailsService.getAllItems();
     }
 
     @PostMapping()
-    public Item createNewItem(@RequestBody ItemDto itemDto){
+    public Item createNewItem(@RequestBody ItemDto itemDto) {
         return itemDetailsService.addNewItem(itemDto);
     }
+
+    @PutMapping("/{id}")
+    public Item putItem(@RequestBody ItemDto itemDto, @PathVariable String id) {
+        Item itemToChange = Item.builder()
+                .id(id)
+                .name(itemDto.getName())
+                .build();
+        return itemDetailsService.changeItem(itemToChange);
+    }
```

Comment:
Ist es sinvoller solche Aufgaben in der Service Klasse durchführen zulassen? Zwecks besserer Unit-Tests

Reply:
Hm, beides ist sinnvoll. Jaaaaa ich mache mir ein Todo zum Service Klassentest schreiben. ;)

Reply:
Jaaaa, in beidem ist es sinnvoll. Das eine ist mehr integriert, dass andere läuft dann mehr auf Unittestebene. Ich mache mir eine Aufgabe zum Nacharbeiten.